p1-staking-pool-dyn/src/lib.rs

```diff
@@ -150,7 +138,8 @@ impl Contract {
     pub fn unstake(&mut self, amount: U128) -> Promise {
         assert_one_yocto();
         let amount = u128::from(amount);
-        let (aid, mut vault) = self.get_vault();
+        let aid = env::predecessor_account_id();
+        let mut vault = self.get_vault_or_default(&aid);
```

Comment:
Why changing this? We should return default - we should panic if the vault doesn't exist.

Reply:
that's one of the Eugene's recommendation, to have a fn get_vault_or_default
I've added a check to panic if vault.staked==0